README.md

```diff
@@ -25,16 +25,87 @@ representing links. For example:
 More detail about HAL can be found at
 [http://stateless.co/hal_specification.html](http://stateless.co/hal_specification.html).
 
-Instructions
-============
+# Instructions
+
 All you should need to do is copy the files into your webroot.
 It is OK to put it in a subdirectory; it does not need to be in the root.
 
 All the JS and CSS dependencies come included in the vendor directory.
 
+# Customizing Options
+
+## Altering the non-GET dialog
+
+Out-of-the-box, the HAL Browser allows users to create new resources through a dialog box, as shown below:
+
+![Default non-GET dialog](images/default-non-get-dialog.png "Default non-GET dialog has simple *Body* input")
+
+Because the HAL Browser has no metadata to describe the resources, it requires the user to enter a raw JSON document in the `Body` field. This requires users to know all the right fields.
+
+If you have some form of metadata (such as an ALPS micro-document), you can plug in your own logic and render a more user-friendly dialog with each attribute listed.
+
+The following fragment of JavaScript shows the API to plugin your own solution:
+```javascript
+HAL.customPost = {
+
+	renderForm: function(href, callback) {
+		// 1. var results = /* array of attribute names ["firstName", "lastName"] */
+		// 2. invoke callback(results)
```

Comment:
It feels like this could be more general ie. the callback could take a string of html